doc/api/assert.markdown

```diff
@@ -2,42 +2,130 @@
 
     Stability: 3 - Locked
 
-This module is used so that Node.js can test itself. It can be accessed with
-`require('assert')`. However, it is recommended that a userland assertion
-library be used instead.
+While the `assert` module is generally intended for internal use by Node.js
+itself, it can be used by user code calling `require('assert')`.
+
+The API for the `assert` module is [Locked][]. This means that there will be no
+additions or changes to any of the methods implemented and exposed by
+the module. It is recommended that an alternative ("userland" or third-party)
+assertion module be used instead.
```

Comment:
I assert ;-) that no such modules exist for the purpose for which assert is well-suited, and arguably intended, in-source assertions of invariants.

Can you point to examples of userland modules for this purpose, @jasnell?

Pointing people writing unit tests elsewhere is a good idea, there are lots of unit test frameworks around.